src/keycard/keycard.cljs

```diff
@@ -254,17 +256,19 @@
         (catch on-failure))))
 
 (defn sign
-  [{pin :pin path :path card-hash :hash on-success :on-success on-failure :on-failure}]
-  (when (and pin card-hash)
-    (if path
-      (.. status-keycard
-          (signWithPath pin path card-hash)
-          (then on-success)
-          (catch on-failure))
-      (.. status-keycard
-          (sign pin card-hash)
-          (then on-success)
-          (catch on-failure)))))
+  [{:keys [pin path hash-data]}]
+  (if path
+    (oops/ocall status-keycard "signWithPath" pin path hash-data)
+    (oops/ocall status-keycard "sign" pin hash-data)))
```

Comment:
@flexsurfer changed this function because we need to return a promise, so that we can sign multiple hashes with `Promise.all`